### ISSUE TITLE
UploadService - Preventing upload! An upload with the same ID is already in progress. Every upload must have unique ID. Please check your code and fix it!

### ISSUE THREAD

**Issue by tobias-r** [gotev/android-upload-service#586]
Hi there,

I am getting the following error message on any multipart upload:

UploadService - Preventing upload! An upload with the same ID is already in progress. Every upload must have unique ID. Please check your code and fix it!

However, nothing is prevented, the upload still starts and works fine. I am just irritated from the message.

Here is my code, it is not setting the upload ID. Even if I do with a random string, the error messages shows.

```
UploadNotificationConfig config = new UploadNotificationConfig(NotificationManagerHelper.NOTIFICATION_CHANNEL_ID_UPLOAD,false,progress,success,error,cancelled);

final MultipartUploadRequest request = new MultipartUploadRequest(mContext, IMAGE_UPLOAD_URL).addFileToUpload(XXX.getFilePath(), "file").setNotificationConfig((context, uploadId) -> config);

request.startUpload();
```

Can you help?

Thank you

Tobias

**Comment by gotev** [comment]
File a demo app which replicates the problem 😉

**Comment by gotev** [comment]
Just released 4.5.5 with [some checks](https://github.com/gotev/android-upload-service/commit/3df3f94e0db6e86df310c48677f24444395e66da) to prevent two common errors. Try this version on your current app and let me know what you get when you replicate the scenario you described.

**Comment by tobias-r** [comment]
Hi there, I just found out about the issue.

After creating a MultipartUploadRequest and subscribing to it, the upload immediately starts, even without calling "request.startUpload()", which I did. So the code wanted to trigger the upload twice, which was prevented.

If you ask me, it's nice to start the upload manually through a call, but it doesn't really matter, it works now.

Thanks for your great API

Tobias
